pulsar-client/src/test/java/org/apache/pulsar/client/impl/conf/ClientConfigurationDataTest.java

```diff
@@ -18,38 +18,109 @@
+    }
+
+    @Test
+    public void testSetAuthFromPropsUsingAuthParams() throws PulsarClientException {
+        ClientConfigurationData conf = new ClientConfigurationData();
+        conf.setAuthPluginClassName(BASIC_AUTH_PLUGIN_CLASS);
+        conf.setAuthParams(basicAuthParams("user1", "pass1"));
+        conf.setAuthenticationFromPropsIfAvailable();
+        assertThatConfHasBasicAuth(conf, "user1", "pass1");
+    }
+
+    @Test
+    public void testSetAuthFromPropsUsingAuthParamMap() throws PulsarClientException {
+        ClientConfigurationData conf = new ClientConfigurationData();
+        conf.setAuthPluginClassName(BASIC_AUTH_PLUGIN_CLASS);
+        conf.setAuthParamMap(basicAuthParamMap("user1", "pass1"));
+        conf.setAuthenticationFromPropsIfAvailable();
+        assertThatConfHasBasicAuth(conf, "user1", "pass1");
+    }
+
+    @Test
+    public void testSetAuthFromPropsWithAllPropsAvailable() throws PulsarClientException {
+        ClientConfigurationData conf = new ClientConfigurationData();
+        conf.setAuthPluginClassName(BASIC_AUTH_PLUGIN_CLASS);
+        conf.setAuthParams(basicAuthParams("user1", "pass1"));
+        conf.setAuthParamMap(basicAuthParamMap("user2", "pass2"));
+        conf.setAuthenticationFromPropsIfAvailable();
+        assertThatConfHasBasicAuth(conf, "user1", "pass1");
+    }
+
+    private void assertThatConfHasBasicAuth(ClientConfigurationData conf, String userId, String password) {
+        assertThat(conf.getAuthentication())
+                .isInstanceOf(AuthenticationBasic.class)
+                .hasFieldOrPropertyWithValue("userId", userId)
+                .hasFieldOrPropertyWithValue("password", password);
+    }
+
+    @Test
+    public void testSetAuthFromPropsWithNoPropsAvailable() throws PulsarClientException {
+        ClientConfigurationData conf = new ClientConfigurationData();
+        conf.setAuthenticationFromPropsIfAvailable();
+        assertThatConfHasNoAuth(conf);
+    }
+
+    @Test
+    public void testSetAuthFromPropsWithOnlyPluginClassNameAvailable() throws PulsarClientException {
+        ClientConfigurationData conf = new ClientConfigurationData();
+        conf.setAuthPluginClassName(BASIC_AUTH_PLUGIN_CLASS);
+        conf.setAuthenticationFromPropsIfAvailable();
+        assertThatConfHasNoAuth(conf);
+    }
+
+    @Test
+    public void testSetAuthFromPropsWithNoPluginClassNameAvailable() throws PulsarClientException {
+        ClientConfigurationData conf = new ClientConfigurationData();
+        conf.setAuthParams(basicAuthParams("user1", "pass1"));
+        conf.setAuthenticationFromPropsIfAvailable();
+        assertThatConfHasNoAuth(conf);
+    }
+
+    private void assertThatConfHasNoAuth(ClientConfigurationData conf) {
+        // getAuthentication() returns disabled when null
+        assertThat(conf.getAuthentication()).isInstanceOf(AuthenticationDisabled.class);
+    }
+
+    private String basicAuthParams(String userId, String password) {
```

Comment:
I am aware that `userId/password` is always `user1/pass1`. However, I left it as such for readability in the test cases.  I find that
```java
conf.setAuthParams(basicAuthParams("user1", "pass1"));
assertThatConfHasBasicAuth("user1", "pass1");
```
is easier to resonate w/ than:
```java
conf.setAuthParams(BASIC_AUTH_PARAMS);
assertThatConfHasBasicAuth();
```
especially when the testing is focusing on what happens when we set these differing auth properties.